src/test/java/org/traccar/helper/IpRetrieverTest.java

```diff
@@ -0,0 +1,38 @@
+package org.traccar.helper;
+
+
+import org.junit.Before;
+import org.junit.Test;
+import org.springframework.mock.web.MockHttpServletRequest;
+
+import static org.junit.Assert.assertEquals;
+
+public class IpRetrieverTest {
+
+    private static final String NORMAL_IP_ADDRESS = "231.23.45.65";
+    private static final String GATEWAY_IP_ADDRESS = "147.120.1.5";
+    private static final String IP_ADDRESS_BEHIND_REVERSE_PROXY = "231.23.45.65, 10.20.10.33, 10.20.20.34";
```

Comment:
Doesn't make sense to have constants defined for unit tested values.

Reply:
Perhaps I can define them in the tests where they are being used at but using the values without variables would lead to confusion.

Reply:
I don't really see why it would lead to a confusion. It's pretty obvious what the values are. You can name test cases appropriately to explain what you are testing.